app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java

```diff
@@ -16,85 +19,201 @@
+    Observable<Void> resumeDiscoveryActivity();
+
+    /** Emits the user name to be displayed. */
+    Observable<String> userNameTextViewText();
   }
 
-  public void emptyProfileViewHolderExploreProjectsClicked(final @NonNull EmptyProfileViewHolder viewHolder) {
-    this.showDiscovery.onNext(null);
+  final class ViewModel extends ActivityViewModel<ProfileActivity> implements ProfileAdapter.Delegate, Inputs, Outputs {
+    private final ApiClientType client;
+    private final CurrentUserType currentUser;
+
+    public ViewModel(final @NonNull Environment environment) {
+      super(environment);
+
+      this.client = environment.apiClient();
+      this.currentUser = environment.currentUser();
+
+      final Observable<User> freshUser = this.client.fetchCurrentUser()
+        .retry(2)
+        .compose(Transformers.neverError());
+      freshUser.subscribe(this.currentUser::refresh);
+
+      final DiscoveryParams params = DiscoveryParams.builder()
+        .backed(1)
+        .perPage(18)
+        .sort(DiscoveryParams.Sort.ENDING_SOON)
+        .build();
+
+      final ApiPaginator<Project, DiscoverEnvelope, DiscoveryParams> paginator =
+        ApiPaginator.<Project, DiscoverEnvelope, DiscoveryParams>builder()
+          .nextPage(this.nextPage)
+          .envelopeToListOfData(DiscoverEnvelope::projects)
+          .envelopeToMoreUrl(env -> env.urls().api().moreProjects())
+          .loadWithParams(__ -> this.client.fetchProjects(params))
+          .loadWithPaginationPath(this.client::fetchProjects)
+          .build();
+
+      final Observable<User> loggedInUser = this.currentUser.loggedInUser();
+
+      this.avatarImageViewUrl = loggedInUser.map(u -> u.avatar().medium());
+
+      this.backedCountTextViewHidden = loggedInUser
+        .map(u -> IntegerUtils.isZero(u.backedProjectsCount()));
+      this.backedTextViewHidden = this.backedCountTextViewHidden;
+
+      this.backedCountTextViewText = loggedInUser
+        .map(User::backedProjectsCount)
+        .filter(IntegerUtils::isNonZero)
+        .map(NumberUtils::format);
+
+      this.createdCountTextViewHidden = loggedInUser
+        .map(u -> IntegerUtils.isZero(u.createdProjectsCount()));
+      this.createdTextViewHidden = this.createdCountTextViewHidden;
+
+      this.createdCountTextViewText = loggedInUser
+        .map(User::createdProjectsCount)
+        .filter(IntegerUtils::isNonZero)
+        .map(NumberUtils::format);
+
+      this.dividerViewHidden = Observable.combineLatest(
+        this.backedTextViewHidden,
+        this.createdTextViewHidden,
+        Pair::create
+      )
+        .map(p -> p.first || p.second);
+
+      this.projects = paginator.paginatedData();
+      this.resumeDiscoveryActivity = this.exploreProjectsButtonClicked;
+      this.startProjectActivity = this.projectCardClicked;
+      this.userNameTextViewText = loggedInUser.map(User::name);
+
+      this.koala.trackProfileView();
```

Comment:
i suppose this is ok for now, but i think in general we only want to trigger this from a user event. in ios we often base this off of `viewDidLoad` or `viewDidAppear`. the equivalent in android might be when the `intent` is configured or something like that. something for us to discuss soon!